client/src/containers/InsertLinkModal/InsertLinkModal.js

```diff
@@ -93,7 +93,7 @@ const createInsertLinkModal = (sectionConfigKey, formName) => {
 
     // get the schemaUrl to use as a key for overrides
     const schemaUrl = `${sectionConfig.form[formName].schemaUrl}${requireTextFieldUrl}`
-      .replace(/:pageid/, ownProps.currentPageID);
+      .replace(/:pageid/, ownProps.currentPageID || 0);
```

Comment:
Most link forms don't actually need the current page, so the JS plugins don't pass it through. In those cases `0` is sufficient.

Without this it'd be `undefined` which gets included as a string in the URL which causes errors.
The alternative would be to update the other link plugins to include the actual ID but a) the way the anchor link plugin grabs it right now feels a little flakey and b) they don't need it so there's no need to add that functionality for them.